tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -3944,23 +3958,31 @@ Status ConvertGather(OpConverterParams* params) {
   int trt_axis = 0;
   TF_RETURN_IF_ERROR(ConvertAxis(axis[0], inputs.at(0).GetTrtDims().nbDims,
                                  node_def.name(), &trt_axis));
-  const TRT_TensorOrWeights& params_tensor = inputs.at(0);
   const TRT_TensorOrWeights& indices_tensor = inputs.at(1);
   if (indices_tensor.batch_size() != 1) {
     return errors::InvalidArgument("Only indices with batch 1 are supported.");
   }
   // Both input are tensors, and the TF gather result will have rank:
   // (params.nbDims + 1) + (indices.nbDims + 1) - 1,
   // where "+ 1" adds the batch dim.
-  const int tf_gather_output_rank = params_tensor.GetTrtDims().nbDims +
-                                    indices_tensor.GetTrtDims().nbDims + 1;
+  const int tf_gather_output_rank =
+      inputs.at(0).GetTrtDims().nbDims + indices_tensor.GetTrtDims().nbDims + 1;
   if (tf_gather_output_rank > nvinfer1::Dims::MAX_DIMS + 1) {
     return errors::InvalidArgument(
         "Result of gather has dimension greater than ",
         nvinfer1::Dims::MAX_DIMS + 1);
   }
   if (params->validation_only) return Status::OK();
+  nvinfer1::ITensor* params_input;
+  // convert weights to tensor
+  if (inputs.at(0).is_tensor()) {
+    params_input = inputs.at(0).tensor();
+  } else {
+    params_input = params->converter->CreateConstantLayer(
```

Comment:
My understanding is, this will add an implicit batch dim of size 1 to the result ITensor, that means the output will have different shape than when the input is a tensor. Say `params` is a tensor and we have `params_shape_with_batch_dim=[1,2,3], index_shape_with_batch_dim=[1,1], axis=2`, then we'll have `output_shape=[1,2,1,1]`. But if `params` is weights, then with the same shape information and with this PR, the params will become an ITensor of shape (including batch dim) `[1,1,2,3]`, then `output_shape` will be `[1,1,1,3]`?

Reply:
Thanks for your comment!
You are right.
In case of using weights I will just add one to it.

Reply:
Thank you for your comment!
You are right. We don't need to subtract 1 from the axis when `params` is a weight.
We might also have some problem when the `axis` is different than 0 (because of batch dimension).
For example, when we have `params_shape=[20, 1, 2, 3], axis=3 and params=[N]`, where `N` is trt batch size, the output should be `[20, 1, 2, N]` (tensorrt currently cannot handle that). 
I will add some checker for that.

Reply:
@aaroey could you review the changes?